### ISSUE TITLE
SoftArgMax is not SoftArgMax

### ISSUE THREAD

**Issue by frankseide** [pytorch/pytorch#104809]
### 🐛 Describe the bug

The function `softargmax` inside the quantized folder (`aten/src/ATen/native/quantized/cpu/qnnpack/src/softargmax.c`) is incorrectly named. It does not seem to compute soft argmax, but just softmax. These are very different functions.

This is corroborated by `aten/src/ATen/native/quantized/cpu/qsoftmax.cpp`, which calls this function to implement softmax.

This can also be seen in the test code, `aten/src/ATen/native/quantized/cpu/qnnpack/test/softargmax-operator-tester.h`, which clearly computes an `outputRef` via the softmax formula. I confirmed this by single-stepping through it.

Could this naming issue please be fixed, so that folks don't spend time on this? Thanks!

### Versions

Current master branch.

**Comment by ptrblck** [comment]
> It does not seem to compute soft argmax, but just softmax. These are very different functions.

That's an interesting statement, as I assumed `softargmax` is just the more precise naming for `softmax` according to [Deep Learning - Goodfellow-et-al-2016](https://www.deeplearningbook.org/contents/mlp.html):
> The name “softmax” can be somewhat confusing. The function is more closely related to the arg max function than the max function. The term “soft” derives from the fact that the softmax function is continuous and diﬀerentiable. The arg max function, with its result represented as a one-hot vector, is not continuous or diﬀerentiable. The softmax function thus provides a “softened” version of the arg max. The corresponding soft version of the maximum function is `softmax(z)^T z`.
It would perhaps be better to call the softmax function “softargmax,” but the current name is an entrenched convention.

Could you describe the difference between these methods based on your understanding?

PS: I'm not the code owner of the quantization backend, but just saw the issue and would like to clarify my potential misunderstanding.

**Comment by jerryzh168** [comment]
cc @kimishpatel @digantdesai

**Comment by mcr229** [comment]
it does seem like the softmax function may often also be referred to as softargmax. I'm fine with keeping the naming as we have it now. If there is truly some disagreement with this, feel free to reopen this issue and I will put up a PR
